orderer/common/cluster/clusterservice_test.go

```diff
@@ -0,0 +1,427 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package cluster_test
+
+import (
+	"context"
+	"crypto/tls"
+	"io"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/hyperledger/fabric-protos-go/common"
+	"github.com/hyperledger/fabric-protos-go/msp"
+	"github.com/hyperledger/fabric-protos-go/orderer"
+	"github.com/hyperledger/fabric/common/flogging"
+	"github.com/hyperledger/fabric/common/metrics/disabled"
+	"github.com/hyperledger/fabric/internal/pkg/comm"
+	"github.com/hyperledger/fabric/orderer/common/cluster"
+	"github.com/hyperledger/fabric/orderer/common/cluster/mocks"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/peer"
+)
+
+var (
+	nodeAuthrequest = &orderer.ClusterNodeServiceStepRequest{
+		Payload: &orderer.ClusterNodeServiceStepRequest_NodeAuthrequest{
+			NodeAuthrequest: &orderer.NodeAuthRequest{
+				Channel: "mychannel",
+			},
+		},
+	}
+	nodeConsensusRequest = &orderer.ClusterNodeServiceStepRequest{
+		Payload: &orderer.ClusterNodeServiceStepRequest_NodeConrequest{
+			NodeConrequest: &orderer.NodeConsensusRequest{
+				Payload: []byte{1, 2, 3},
+			},
+		},
+	}
+	nodeTranRequest = &orderer.ClusterNodeServiceStepRequest{
+		Payload: &orderer.ClusterNodeServiceStepRequest_NodeTranrequest{
+			NodeTranrequest: &orderer.NodeTransactionOrderRequest{
+				LastValidationSeq: 0,
+				Payload:           &common.Envelope{},
+			},
+		},
+	}
+	nodeInvalidRequest = &orderer.ClusterNodeServiceStepRequest{
+		Payload: &orderer.ClusterNodeServiceStepRequest_NodeConrequest{
+			NodeConrequest: nil,
+		},
+	}
+	submitRequest = &orderer.StepRequest{
+		Payload: &orderer.StepRequest_SubmitRequest{
+			SubmitRequest: &orderer.SubmitRequest{
+				LastValidationSeq: 0,
+				Payload:           &common.Envelope{},
+				Channel:           "mychannel",
+			},
+		},
+	}
+)
+
+type AuthVerifier struct {
+	cluster.ClusterService
+}
+
+func (cs *AuthVerifier) VerifyAuthRequest(orderer.ClusterNodeService_StepServer, *orderer.ClusterNodeServiceStepRequest, cluster.TLSSessionBindingGetter, cluster.SignatureVerifier) (string, error) {
+	return "mychannel", nil
+}
+
+func TestClusterServiceStep(t *testing.T) {
```

Comment:
can we please put t.Parallel() on all tests? so they will run together, less time.